Define dependency maintenance policy (incl. Spring) (#5498)
camunda/camunda-docs: docs/reference/supported-environments.md

</TabItem>

</Tabs>

## Dependency maintenance policies

> Camunda provides [a standard support policy](https://camunda.com/release-policy/) of 18 months for a particular minor version, from the date it is released.

Reviewer: ```suggestion
Camunda provides [a standard support policy](https://camunda.com/release-policy/) of 18 months for a particular minor version from the date it is released.
```

> During this time, patches are regularly released which contain security and bug fixes, some of which may come from dependency updates. As such, for the

Reviewer: ```suggestion
During this time, patches are regularly released containing security and bug fixes, some of which may come from dependency updates. Therefore, the
```

vast majority of dependencies, Camunda _only_ applies patch updates.

> Certain dependencies used by Camunda 8, however, may have a shorter maintenance policy than Camunda itself. For those dependencies which are integral to its functionality, Camunda may adopt a different update policy, as listed below.

Reviewer: ```suggestion
However, certain dependencies used by Camunda 8 may have a shorter maintenance policy than Camunda itself. Camunda may adopt a different update policy for these dependencies, as listed below.
```

### Spring

> Many server side components, such as the **Camunda Orchestration Cluster**, are Spring Boot applications, and leverage Spring Boot (and the wider Spring ecosystem) to execute fundamental functionality such as application configuration, REST infrastructure (including security), production ready features, etc.

Reviewer: This is a good place to link to the Camunda Orchestration Cluster, as defined in the glossary. We should likely make it very clear how we have defined this for both current and new users, but I just realized we haven't added this to the glossary yet.

I would propose we provide a link to this glossary definition in every instance of Camunda Orchestration Cluster for now -- additionally, is this the casing we want to standardize on? CC @MaxTru @akeller 

@npepinpe This is out of scope for this PR, so you can disregard.

Reviewer 2: We are still working on some of the terminology in 8.8, but I'm pretty confident this will just be "Orchestration cluster" with no "Camunda" qualifier on the front. I definitely think it's a good opportunity in 8.8 with the intro of the Orchestration cluster that we link to the glossary definition.

Reviewer 2: Related - https://github.com/camunda/camunda-docs/pull/5537

> However, Spring has a shorter maintenance window than Camunda for its open-source software (OSS) offering. Versions are only supported for 12 months, versus Camunda's 18 months. To circumvent this, server-side components of Camunda 8 patch releases also update Spring (and Spring Boot) minor versions, such that the latest patch release of these components uses a supported Spring version.

Reviewer 3: You could link here to the support policy documentation for [Spring](https://spring.io/projects/spring-framework#support) and [Spring Boot](https://spring.io/projects/spring-boot#support). In that sense, 12 months for Spring is not fully accurate (it's Spring Boot, Spring Framework tends to have longer support durations). Maybe you can clarify that or make the statement more generic (like "Spring and Spring Boot minor versions can go out of maintenance during the maintenance window of a Camunda minor release").

Reviewer 4: @npepinpe Can we explicitly limit it to the orchestration cluster?

Author: Spring support is a little hard to explain, since it's tied to the Spring Boot version. But I can keep it to Spring Boot for now.

> All project support dates are mapped to the version of Spring Boot that either manages it or it depends upon (the support dates for a given project will match those of the Spring Boot version it’s mapped to).

Linking is a good idea :+1:

> Spring, however, has a shorter maintenance window than Camunda for its OSS offering: versions are only supported for 12 months, versus Camunda's 18 months. To circumvent this, server side components of Camunda 8 patch releases also update Spring (and Spring Boot) minor versions, such that the latest patch release of these components uses a supported Spring version.

Author: I applied the suggestion anyway, but changed the OSS definition.

> As for libraries and SDKs meant to be included in third-party applications, Camunda follows a best effort policy to balance compatibility and securing Spring-dependent libraries.

Reviewer: ```suggestion
As for libraries and SDKs meant to be included in third-party applications, Camunda follows a best effort policy to balance compatibility and secure Spring-dependent libraries.
```
